packages/typespec-azure-resource-manager/src/rules/no-empty-model.ts

```diff
@@ -14,7 +14,7 @@ export const noEmptyModel = createRule({
   create(context) {
     return {
       model: (model: Model) => {
-        if (model.properties.size === 0) {
+        if (model.properties.size === 0 && model.sourceModel?.name !== "Record") {
```

Comment:
We should use the built-in compiler call for `isRecordModelType` as in this example: https://github.com/Azure/typespec-azure/blob/60f9e0682a8b3ff7e03fde33a2a199053db7d1d7/packages/typespec-autorest/src/openapi.ts#L1903

see import here: https://github.com/Azure/typespec-azure/blob/60f9e0682a8b3ff7e03fde33a2a199053db7d1d7/packages/typespec-autorest/src/openapi.ts#L87